payments_redsys/__init__.py

```diff
@@ -1,259 +1,292 @@
-                    payment.captured_amount = int(merchant_parameters['Ds_Amount']) / 100
-                    payment.transaction_id = merchant_parameters['Ds_AuthorisationCode']
-                    payment.extra_data = merchant_parameters
-                    payment.change_status('confirmed')
-                    logger.debug('payment %d confirmed' % payment.pk)
-                elif transaction_type == '1':
-                    payment.extra_data = merchant_parameters
-                    payment.change_status('preauth')
-                    logger.debug('payment %d preauthorised' % payment.pk)
-                else:
-                    logger.debug('authorised payment response but unrecognised transaction type %s' % transaction_type)
-
-            if response_code == 900:
-                # Authorised transaction for refunds and confirmations
-                if transaction_type == '3':
-                    payment.extra_data = merchant_parameters
-                    payment.change_status('refunded')
-                    logger.debug('payment %d automatic refund' % payment.pk)
-                else:
-                    logger.debug('authorised refund response but unrecognised transaction type %s' % transaction_type)
-
-            if response_code > 100 and response_code != 900:
-                # any of a long list of errors/rejections
-                payment.extra_data = merchant_parameters
-                payment.change_status('rejected', message='Ds_Response was %d' % response_code)
-                # perhaps import and raise PaymentError from django-payments
-                logger.debug('rejected: %s' % binary_merchant_parameters.decode())
-
-        return HttpResponse()
+# Copyright (C) 2018 AJ Ostergaard
+#
+# This file is part of django-payments-redsys.
+#
+# django-payments-redsys is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# django-payments-redsys is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with django-payments-redsys.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import base64
+import hashlib
+import hmac
+import json
+import logging
+import re
+
+import pyDes
+import requests
+from django import forms
+from django.http import HttpResponseRedirect
+from payments import PaymentError
+from payments.core import BasicProvider, get_base_url, urljoin
+from payments.forms import PaymentForm
+
+logger = logging.getLogger(__name__)
+
+# https://en.wikipedia.org/wiki/ISO_4217
+ISO_CURRENCY_LOOKUP = {
```

Comment:
I found it much more convenient to be able to use the currency codes everyone knows... but only added a few in here. Anyone can go ahead and add more.